### DISCUSSION TITLE
Prepared statement and named parameters

### DISCUSSION THREAD

**Discussion by unconverged** [duckdb/duckdb#4142] [Q&A]
Hello!

Does DuckDb support named parameters for prepared statements? I mean, can I use something like `SELECT ?test;`? Or only params with integral indices are supported, like `SELECT $1;` or `SELECT ?;`?

**Comment by Alex-Monahan** [comment]
Hello! Which DuckDB client are you using? (Python, node, CLI, etc.)

**Reply by unconverged** [reply]
I'm talking about low-level [C API](https://duckdb.org/docs/api/c/prepared). For example, SQLite does have named parameters and has an API to get parameter index by name (`sqlite3_bind_parameter_index`). I do not see anything similar for DuckDB.

**Reply by karlb** [reply]
I'm using python and would like to see the "named" paramstyle (e.g. `SELECT * WHERE id=:id`, as described in https://peps.python.org/pep-0249/#paramstyle). This would also increase the compatibility with python's builtin sqlite3.

**Reply by Mytherin** [reply]
This should already be supported in the development version - see #5611

**Reply by karlb** [reply]
Looks great, thanks!

**Reply by Giorgi** [reply]
@Mytherin @Tishj Is the PR for Python only? It would be great to have support for named parameters in C API too.

**Reply by Mytherin** [reply]
It's implemented in the C++ layer, happy to review a PR that adds the appropriate C bindings.
